lnwire/onion_error.go

```diff
@@ -146,6 +147,11 @@ func (c FailCode) String() string {
 // NOTE: May be returned by any node in the payment route.
 type FailInvalidRealm struct{}
 
+func (f *FailInvalidRealm) Error() string {
+	var code = f.Code()
```

Comment:
Can be simplified to:
```go
return f.Code.String()
```

Also in the case that the error has an internal attribute, that should be factored into the returned error message itself.

Reply:
The two comments above apply to all other instances in this diff (not commenting on each of them to avoid review clutter).